src/Symfony/Component/Serializer/Deserialize/Template/EagerTemplateGenerator.php

```diff
@@ -0,0 +1,269 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Deserialize\Template;
+
+use Symfony\Component\Serializer\Deserialize\DataModel\CollectionNode;
+use Symfony\Component\Serializer\Deserialize\DataModel\DataModelNodeInterface;
+use Symfony\Component\Serializer\Deserialize\DataModel\ObjectNode;
+use Symfony\Component\Serializer\Deserialize\DataModel\ScalarNode;
+use Symfony\Component\Serializer\Deserialize\Decoder\DecoderInterface;
+use Symfony\Component\Serializer\Exception\UnexpectedValueException;
+use Symfony\Component\Serializer\Php\ArgumentsNode;
+use Symfony\Component\Serializer\Php\ArrayAccessNode;
+use Symfony\Component\Serializer\Php\ArrayNode;
+use Symfony\Component\Serializer\Php\AssignNode;
+use Symfony\Component\Serializer\Php\BinaryNode;
+use Symfony\Component\Serializer\Php\CastNode;
+use Symfony\Component\Serializer\Php\ClosureNode;
+use Symfony\Component\Serializer\Php\ExpressionNode;
+use Symfony\Component\Serializer\Php\ForEachNode;
+use Symfony\Component\Serializer\Php\FunctionCallNode;
+use Symfony\Component\Serializer\Php\IfNode;
+use Symfony\Component\Serializer\Php\MethodCallNode;
+use Symfony\Component\Serializer\Php\NewNode;
+use Symfony\Component\Serializer\Php\ParametersNode;
+use Symfony\Component\Serializer\Php\PhpNodeInterface;
+use Symfony\Component\Serializer\Php\ReturnNode;
+use Symfony\Component\Serializer\Php\ScalarNode as PhpScalarNode;
+use Symfony\Component\Serializer\Php\ThrowNode;
+use Symfony\Component\Serializer\Php\TryCatchNode;
+use Symfony\Component\Serializer\Php\VariableNode;
+use Symfony\Component\Serializer\Php\YieldNode;
+
+/**
+ * Generates a template PHP syntax tree that deserializes data eagerly.
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @experimental in 7.0
+ */
+final class EagerTemplateGenerator extends TemplateGenerator
+{
+    /**
+     * @param class-string<DecoderInterface> $decoderClassName
+     */
+    public function __construct(
+        private readonly string $decoderClassName,
+    ) {
+    }
+
+    protected function returnDataNodes(DataModelNodeInterface $node, array &$context): array
+    {
+        return [
+            new ExpressionNode(new ReturnNode(new FunctionCallNode(
```

Comment:
Is using an AST like this (instead of generating PHP files directly as we do in the DI component) really necessary?

If that's the case, I'm also in favor of using Nikic PHP Parser to reduce the complexity and the maintenance burden.

Reply:
How much confident are we about the compatibility of nikic/php-parser's release cycle and ours? En

Reply:
For using php-parser in some of our project it's really stable, version 4 was released 5 years ago, and there is a new version upcoming (5.0) already available in beta (and from what i have read in https://github.com/nikic/PHP-Parser/blob/master/UPGRADE-5.0.md there may be no changed needed between 4 and 5 when doing code generation)

Reply:
Also for generating code with AST or not the main pro in my view are : 

 * Text: no need for dependencies, "easy" for small generation
 * AST: better to catch errors, better when generation is done in different code blocks (since you can type checking what a function need to correctly handle generation : a variable ? a method call ? a assignment ? ....) in text mode you only have string and so you need to be aware of all the implementation detail

Reply:
I think the AST approach is far superior, when you have to adapt to a new PHP version you only fix it once. It's also much more unit testable. If you intend to generate code at many places I think having the dependency well worth the burden.

Reply:
Moreover, I'm doing an optimization step within compilation (basically merging fwrites). This won't be possible with text, or way too tricky.

Reply:
I removed the custom AST implementation in favor of `nikic/php-parser`, I indeed think that we can rely on that stable library and remove the amount of code here.